crates/signature-mpc/src/twopc_mpc_protocols/dwallet_transfer.rs

```diff
@@ -0,0 +1,361 @@
+        witnesses,
+        &mut OsRng,
+    )
+    .unwrap();
+
+    ProofPublicOutput {
+        proof: proofs,
+        encrypted_discrete_log: statements[0]
+            .language_statement()
+            .encrypted_discrete_log()
+            .value(),
+        range_proof_commitment: statements[0].range_proof_commitment().value(),
+    }
+}
+
+fn generate_witnesses(
+    randomness: RandomnessSpaceGroupElement,
+    enhanced_language_public_parameters: &language::PublicParameters<
+        { maurer::SOUND_PROOFS_REPETITIONS },
+        EnhancedLang<
+            { maurer::SOUND_PROOFS_REPETITIONS },
+            RANGE_CLAIMS_PER_SCALAR,
+            RandomnessSpaceGroupElement,
+            Lang,
+        >,
+    >,
+    plaintext: PlaintextSpaceGroupElement,
+) -> Vec<
+    WitnessSpaceGroupElement<
+        { maurer::SOUND_PROOFS_REPETITIONS },
+        RANGE_CLAIMS_PER_SCALAR,
+        COMMITMENT_SCHEME_MESSAGE_SPACE_SCALAR_LIMBS,
+        bulletproofs::RangeProof,
+        RandomnessSpaceGroupElement,
+        Lang,
+    >,
+> {
+    let witnesses: Vec<language::WitnessSpaceGroupElement<1, Lang>> =
+        vec![(plaintext, randomness).into()];
+
+    let witnesses =
+        EnhancedLanguage::<
+            { maurer::SOUND_PROOFS_REPETITIONS },
+            RANGE_CLAIMS_PER_SCALAR,
+            { COMMITMENT_SCHEME_MESSAGE_SPACE_SCALAR_LIMBS },
+            bulletproofs::RangeProof,
+            RandomnessSpaceGroupElement,
+            Lang,
+        >::generate_witnesses(witnesses, &enhanced_language_public_parameters, &mut OsRng)
+        .unwrap();
+    witnesses
+}
+
+pub fn enhanced_language_public_parameters<
+    const REPETITIONS: usize,
+    const NUM_RANGE_CLAIMS: usize,
+    UnboundedWitnessSpaceGroupElement: group::GroupElement + Samplable,
+    Lang: EnhanceableLanguage<
+        REPETITIONS,
+        NUM_RANGE_CLAIMS,
+        COMMITMENT_SCHEME_MESSAGE_SPACE_SCALAR_LIMBS,
+        UnboundedWitnessSpaceGroupElement,
+    >,
+>(
+    unbounded_witness_public_parameters: UnboundedWitnessSpaceGroupElement::PublicParameters,
+    language_public_parameters: Lang::PublicParameters,
+) -> language::PublicParameters<
+    REPETITIONS,
+    EnhancedLang<REPETITIONS, NUM_RANGE_CLAIMS, UnboundedWitnessSpaceGroupElement, Lang>,
+> {
+    MaurerPublicParameters::new::<
+        bulletproofs::RangeProof,
+        UnboundedWitnessSpaceGroupElement,
+        Lang,
+    >(
+        unbounded_witness_public_parameters,
+        bulletproofs::PublicParameters::default(),
+        language_public_parameters,
+    )
+    .unwrap()
+}
+pub fn is_valid_proof(
+    language_public_parameters: LangPublicParams,
+    proof_public_output: ProofPublicOutput,
+    centralized_public_keyshare: group::Value<secp256k1::GroupElement>,
+) -> bool {
+    let secp256k1_group_public_parameters = secp256k1::group_element::PublicParameters::default();
+    let protocol_public_parameters = ProtocolPublicParameters::new(DUMMY_PUBLIC_KEY);
```

Comment:
No no no, don't instantiate this struct with a wrong public key. This cannot be a part of production code - you should have been alarmed by the use of the word `DUMMY` inside a production non-test function.

Reply:
I agree that the name `DUMMY` should not be in the production code, but it used to be `N` in many parts of the production code.